routing/ruler_router.cpp

```diff
@@ -100,8 +100,8 @@ RouterResultCode RulerRouter::CalculateRoute(Checkpoints const & checkpoints,
     else
```

Comment:
- Probably: ```Note: endSegmentIdx is decreased``` _is_
- take out common ```subroutes.emplace_back(ToPointWA(points[i-1]), ToPointWA(points[i]), i*2-2, i*2);``` in both conditions.

Reply:
Looks like we can remove if-else branching here. Can't remember why it was introduced. Tested on iOS simulator and Android - ruler works as expected